pkg/client/restclient/negotiated_codec.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restclient
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// FIXME: This required some work.
+// FIXME: This should use versioning.NewCodecForScheme to get encoder and decoder.
+// TODO: This should probably be renamed.
+type negotiatedSerializerWrapper struct {
```

Comment:
This probably belongs in the runtime/serializer package as a component, and we should figure out what happens when someone asks encoder for version and it conflicts with the raw serializer.  That's possibly another reason to continue accepting codec for now.